### ISSUE TITLE
Query triggered with `fetchMore` is retriggered without pagination params and right result is overwritten

### ISSUE THREAD

**Issue by Varpusparvi** [apollographql/apollo-client#11427]
### Issue Description

We are having a problem where next page is fetched with `fetchMore` with `after` param for pagination. The query finishes correctly and is added to the cache, but immediately after that pagination info in the cache is reset to the previous page's info.

As far as I can see, here we get the right result in `fetchMoreResult` but then we update the cache / trigger subscribes with the `currentResult` (which comes from `getCurrentResult`) is now the old page.
```
  function fetchMore (options: FetchMoreQueryOptions<TVariables, TResult> & FetchMoreOptions<TResult, TVariables>) {
    if (query.value) {
      error.value = null
      loading.value = true
      return query.value.fetchMore(options)
        .then((fetchMoreResult) => {
          const currentResult = query.value?.getCurrentResult()
          currentResult && processNextResult(currentResult)
          return fetchMoreResult
        })
    }
  }
```

<img width="1237" alt="Screenshot 2023-12-12 at 13 09 24" src="https://github.com/apollographql/apollo-client/assets/44475420/d03784e0-db58-4ae0-9ef3-db2461556c90">

Is it correct that the `processNextResult` is fired with the _old result_ instead of the one from `fetchMoreResult`?

In other words:

We fire `fetchMoreResult` and it is fired correctly and has the right response, but the result is not propagated to the `useQuery` correctly causing retriggering of the query without pagination params and this is then propagated causing UI issues because it's old/partial data from wrong page now

### Link to Reproduction

Couldn't do it yet

**Comment by bignimbus** [comment]
Hi @Varpusparvi 👋🏻  thanks for letting us know about this!  Unfortunately it’s difficult to know what the issue might be without being able to run the code that you’re sharing.  I recommend sharing a [runnable reproduction](https://www.apollographql.com/blog/apollo-client/community-update-apollo-client-apollo-ios-and-apollo-kotlin/#share-as-many-details-as-possible) so the community and maintainers can help narrow down whether this is an issue with your application code or with Apollo Client.  Looking forward to hearing back from you 🙏🏻
